src/platform_impl/linux/x11/util/randr.rs

```diff
@@ -107,16 +100,50 @@ impl XConnection {
             (*output_info).nameLen as usize,
         );
         let name = String::from_utf8_lossy(name_slice).into();
-        let hidpi_factor = if let Some(dpi) = self.get_xft_dpi() {
-            dpi / 96.
-        } else {
-            calc_dpi_factor(
+        // Override DPI if `WINIT_HIDPI_FACTOR` variable is set
+        let dpi_env = env::var("WINIT_HIDPI_FACTOR").ok().map_or_else(
+            || EnvVarDPI::NotSet,
+            |var| {
+                if var.to_lowercase() == "randr" {
+                    EnvVarDPI::Randr
+                } else if let Ok(dpi) = f64::from_str(&var) {
+                    EnvVarDPI::Scale(dpi)
+                } else {
+                    EnvVarDPI::NotSet
```

Comment:
Should we blow up here? I think like it makes sense not to do so for backwards compatibility.

Reply:
Why do you say we shouldn't? Winit blew up before if you set the environment variable to an invalid value, and this doesn't break any previously-invalid values.

Reply:
Not really, we allowed arbitrary string before.

Reply:
Oops, didn't see the existing comment chain for this.

It seems that the previous implementation was also silently ignoring any string that didn't parse to a valid `f64`, but I don't think there's any reason to continue supporting that behavior. winit should panic on an invalid value in this env var. After all, we are already panicking on a valid `f64` that constitutes an invalid DPI factor.

Reply:
I think I'm in favor of exploding as a downstream. I'd much rather have it blow up in the face of my users than having to troubleshoot for hours why it doesn't do what it's supposed to do only because there's a zero width character or some funkyness somewhere.

Reply:
Oh, wait, let's at least not blow up on empty string. What do you think?

Reply:
Yes, I think it's reasonable to treat an empty var and an absent var as the same.